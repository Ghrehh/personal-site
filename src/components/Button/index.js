// @flow
import * as React from 'react';

import * as childrenAndClassName from 'models/children-and-class-name';
import styles from './styles.module.css';

const Button = (props: childrenAndClassName.Type) => (
  <button className={`${styles.button} ${props.className}`}>
    {props.children}
  </button>
);

Button.defaultProps = childrenAndClassName.defaultProps;

export default Button;
