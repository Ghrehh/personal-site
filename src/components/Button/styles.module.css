.button {
  outline: none;
  -webkit-tap-highlight-color: transparent;
  background-color: var(--charcoal);
  box-shadow: var(--box-shadow);
  cursor: pointer;
  color: white;
  border: none;
  padding: 11px 15px;
  border-radius: 3px;
  font-size: 18px;
  font-weight: 500;
  transition: transform 0.5s, filter 0.5s;
}

.button:hover {
  transform: translate(0, -3px);
  filter: brightness(1.3);
}
