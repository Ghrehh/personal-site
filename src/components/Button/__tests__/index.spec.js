import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Button from '../';

describe('Button', () => {
  let testContext = {};
  let component = <Button className="foo">FooBarBaz</Button>;

  beforeEach(() => {
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(component);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
