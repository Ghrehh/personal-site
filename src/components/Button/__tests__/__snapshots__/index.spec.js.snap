// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component correctly 1`] = `
<button
  className="button foo"
>
  FooBarBaz
</button>
`;
